src/pipeline_edsl/to_workflow.ml

```diff
@@ -856,20 +856,35 @@ module Make (Config : Compiler_configuration)
     )
 
   let topiary ?(configuration=Tools.Topiary.Configuration.default)
-      vcf predictor alleles =
-    let v = get_vcf vcf in
-    let reference_build = v#product#reference_build in
+      vcfs predictor alleles =
+    let vs = List.map ~f:get_vcf vcfs in
+    let refs = List.map ~f:(fun v -> v#product#reference_build) vs in
```

Comment:
Hmm, I definitely like your approach much better. Didn't know about the `dedup` before :\